Mathlib/CategoryTheory/Yoneda.lean

```diff
@@ -0,0 +1,530 @@
+  map_injective {X} _ _ _ p := by
+    have t := congr_fun (congr_app p X.unop) (𝟙 _)
+    simpa using congr_arg Quiver.Hom.op t
+#align category_theory.coyoneda.coyoneda_faithful CategoryTheory.Coyoneda.coyoneda_faithful
+
+/-- If `coyoneda.map f` is an isomorphism, so was `f`.
+-/
+theorem isIso {X Y : Cᵒᵖ} (f : X ⟶ Y) [IsIso (coyoneda.map f)] : IsIso f :=
+  isIso_of_fully_faithful coyoneda f
+#align category_theory.coyoneda.is_iso CategoryTheory.Coyoneda.isIso
+
+/-- The identity functor on `Type` is isomorphic to the coyoneda functor coming from `punit`. -/
+def punitIso : coyoneda.obj (Opposite.op PUnit) ≅ 𝟭 (Type v₁) :=
+  NatIso.ofComponents
+    (fun X =>
+      { hom := fun f => f ⟨⟩
+        inv := fun x _ => x })
+    (by aesop_cat)
+#align category_theory.coyoneda.punit_iso CategoryTheory.Coyoneda.punitIso
+
+/-- Taking the `unop` of morphisms is a natural isomorphism. -/
+@[simps!]
+def objOpOp (X : C) : coyoneda.obj (op (op X)) ≅ yoneda.obj X :=
+  NatIso.ofComponents (fun _ => (opEquiv _ _).toIso) fun _ {_Y} _f => rfl
+#align category_theory.coyoneda.obj_op_op CategoryTheory.Coyoneda.objOpOp
+
+end Coyoneda
+
+namespace Functor
+
+/-- A functor `F : Cᵒᵖ ⥤ Type v₁` is representable if there is object `X` so `F ≅ yoneda.obj X`.
+
+See <https://stacks.math.columbia.edu/tag/001Q>.
+-/
+class Representable (F : Cᵒᵖ ⥤ Type v₁) : Prop where
+  /-- `Hom(-,X) ≅ F` via `f` -/
+  has_representation : ∃ (X : _)(f : yoneda.obj X ⟶ F), IsIso f
+#align category_theory.functor.representable CategoryTheory.Functor.Representable
+
+instance {X : C} : Representable (yoneda.obj X) where has_representation := ⟨X, 𝟙 _, inferInstance⟩
+
+/-- A functor `F : C ⥤ Type v₁` is corepresentable if there is object `X` so `F ≅ coyoneda.obj X`.
+
+See <https://stacks.math.columbia.edu/tag/001Q>.
+-/
+class Corepresentable (F : C ⥤ Type v₁) : Prop where
+  /-- `Hom(X,-) ≅ F` via `f` -/
+  has_corepresentation : ∃ (X : _)(f : coyoneda.obj X ⟶ F), IsIso f
+#align category_theory.functor.corepresentable CategoryTheory.Functor.Corepresentable
+
+instance {X : Cᵒᵖ} : Corepresentable (coyoneda.obj X)
+    where has_corepresentation := ⟨X, 𝟙 _, inferInstance⟩
+
+-- instance : corepresentable (𝟭 (Type v₁)) :=
+-- corepresentable_of_nat_iso (op punit) coyoneda.punit_iso
+section Representable
+
+variable (F : Cᵒᵖ ⥤ Type v₁)
+
+variable [F.Representable]
+
+/-- The representing object for the representable functor `F`. -/
+noncomputable def reprX : C :=
+  (Representable.has_representation : ∃ (_ : _)(_ : _ ⟶ F), _).choose
+set_option linter.uppercaseLean3 false 
+#align category_theory.functor.repr_X CategoryTheory.Functor.reprX
+
+/-- The (forward direction of the) isomorphism witnessing `F` is representable. -/
+noncomputable def reprF : yoneda.obj F.reprX ⟶ F :=
+  Representable.has_representation.choose_spec.choose
+#align category_theory.functor.repr_f CategoryTheory.Functor.reprF
+
+/- warning: category_theory.functor.repr_x clashes with
```

Comment:
I think you can remove this comment. It no longer applies, right?

Reply:
Removed